packages/client/__tests__/pages/ResourceIndex.spec.js

```diff
@@ -59,13 +71,103 @@ describe('Test the resource index page', () => {
-        const { debug } = render(
+        render(
             <Router history={history}>
                 <ResourceIndex {...props} />
             </Router>
         )
-
         expect(window.Flamingo.request.get).toHaveBeenCalled()
     })
-    test('should be able to edit a post from the post table', () => {})
+    test('per page button changes the pers page value for the table', () => {})
+    test('can select a row on the resource table', async () => {
+        render(
+            <Router history={history}>
+                <ResourceIndex {...props} />
+            </Router>
+        )
+        await waitFor(() =>
+            fireEvent.click(screen.getAllByTestId('table-row')[0])
+        )
+    })
+    test('can search for values on the resource table', async () => {
+        render(
+            <Router history={history}>
+                <ResourceIndex {...props} />
+            </Router>
+        )
+        await waitFor(() =>
+            expect(screen.queryAllByTestId('table-row')).toHaveLength(2)
+        )
+        fireEvent.change(screen.getByTestId('search-resource'), {
+            target: { value: 'Amapai' },
+        })
+        window.Flamingo = {
+            ...FlamingoMock,
+            request: {
+                get: jest.fn().mockResolvedValue({
+                    data: {
+                        data: [
+                            {
+                                av_cpc: 21,
+                                category: 'angular',
+                                content: 'this is the content for amapai',
+                                created_at: '2020-08-12T21:40:24.000Z',
+                                description:
+                                    'this is the description for amapai',
+                                id: 1,
+                                published_at: '2020-08-11T23:00:00.000Z',
+                                scheduled_for: '2020-08-11T23:00:00.000Z',
+                                title: 'Amapai',
+                                updated_at: '2020-08-12T21:40:24.000Z',
+                                user_id: 2,
+                            },
+                        ],
+                        page: 1,
+                        total: 1,
+                        perPage: 10,
+                        pageCount: 1,
+                    },
+                }),
+            },
+        }
+        await waitFor(() =>
+            expect(screen.queryAllByTestId('table-row')).toHaveLength(1)
+        )
+    })
+    test('table row should show correct amount of data', async () => {
+        render(
+            <Router history={history}>
+                <ResourceIndex {...props} />
+            </Router>
+        )
+        await waitFor(() =>
+            expect(screen.queryAllByTestId('table-row')).toHaveLength(2)
+        )
+    })
+    test('clicking on the delete icon should delete that specific row data', async () => {
```

Comment:
This should also make sure the modal is open. Take a snapshot of the modal content. Then clicking the delete button should make an API call with the correct api content. This should also be asserted. Then also assert that the modal is closed. Also assert the component calls the API again to fetch the new data, assert the component is in loading state, then assert it shows the new data.